docs/Using-SRIOV-and-Clear-Containers.md

```diff
@@ -0,0 +1,265 @@
+# Setup to use SR-IOV with Clear Containers
+
+Single Root I/O Virtualization (SR-IOV) enables splitting a physical device into
+virtual functions (VFs). This enables direct passthrough to virtual machines or
+containers.  For Clear Containers, we have enabled a Container Network Model (CNM)
+plugin and necessary changes in the runtime to detect virtual functions in a 
+container's network namespace to facilitate making use of SR-IOV for network 
+based devices.
+
+## Install the SR-IOV Docker\* plugin
+
+In order to create a network with VFs associated with it which can be passed to
+Clear Containers, it is necessary to install a SR-IOV Docker plugin. The 
+created network will be based on a physical function (PF) device, and can
+create 'n' containers, where 'n' is the number of VFs associated with the 
+Physical Function (PF).
+
+To install the plugin, follow the [plugin installation instructions](https://github.com/clearcontainers/sriov).
+
+
+## Host setup for SR-IOV
+
+You need to have a system with Intel VT-d support. Your kernel needs to have 
+Input-Output Memory Management Unit (IOMMU) and Virtual Function I/O (VFIO) 
+support. `CONFIG_VFIO_NOIOMMU` must to be disabled in the host kernel
+configuration for SR-IOV. You will need to rebuild your host system's kernel in
+order to disable `CONFIG_VFIO_NOIOMMU` in the kernel configuration.
+You may also potentially need to add a PCI override for your Network Interface 
+Controller (NIC). The section below describes how to assess if changes are
+needed, and how to make these changes.
+
+### IOMMU Groups and PCIe Access Control Services
+
+Run the following command to see how the IOMMU groups are setup on your 
+host system. This information details whether or not your NIC is setup 
+appropriately with respect to PCIe Access Control Services (ACS). If the PCI 
+bridge is within the same IOMMU group as your NIC, then this is an indication 
+that either your device doesn't support ACS or that it doesn't share this 
+capability appropriately by default.
+
+```
+$ find /sys/kernel/iommu_groups/ -type l
+```
+
+If all if all is setup properly, you will have the PCI for each ACS enabled 
+NIC port in its own IOMMU group. If you don't see any output when running the above
+command, then you likely need to update your host's kernel configuration.
+
+For more details, see http://vfio.blogspot.com/2014/08/iommu-groups-inside-and-out.html
+
+### Updating the Host Kernel
```

Comment:
Use infinitives for headings: Update the host kernel
Make sure all headings use sentence case: Only the first word and proper nouns (names) are capitalized. 
This applies to all headings.